sql/2022/third-parties/content_encoding_by_content_type.sql

```diff
@@ -0,0 +1,67 @@
+#standardSQL
+#content-encoding by third parties by content-type
+
+WITH requests AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    pageid AS page,
+    url,
+    resp_content_encoding AS content_encoding,
+    CASE
+      WHEN resp_content_type LIKE '%html%' THEN 'html'
+      WHEN resp_content_type LIKE '%xml%' THEN 'xml'
+      WHEN resp_content_type LIKE '%plain%' THEN 'plain'
+      WHEN resp_content_type LIKE '%video%' THEN 'video'
+      WHEN resp_content_type LIKE '%audio%' THEN 'audio'
+      WHEN resp_content_type LIKE '%image%' THEN 'image'
+      WHEN resp_content_type LIKE '%font%' THEN 'font'
+      WHEN resp_content_type LIKE '%css%' THEN 'css'
+      WHEN resp_content_type LIKE '%javascript%' THEN 'javascript'
+      WHEN resp_content_type LIKE '%json%' THEN 'json'
+      ELSE 'other'
+    END AS content_type
```

Comment:
I saw that this used elsewhere (ex: sql/2021/javascript/most_requested_content_type.sql) and I assumed it was done intentionally to aggregate similar types together. I will update the query and review the results.

> Note that the JavaScript is listed as script and JSON type is also listed as a script in this case so can't separate them. Not sure if that's an issue for you?

I think one should be showing `javascript` and the other should be showing `json`.

> I don't think the intent of this query is to actually look at response content type per se. And to be honest, I think what you're getting here is covered by percent_of_third_party_requests_and_bytes_by_category_and_content_type.sql

The query you reference shows the `content-type` by 3P category. This query shows the `content-encoding` by `content-type` for 3Ps. For example, are 3Ps compressing images? Are 3Ps using `gzip` for JavaScript?

Reply:
Using the `mimeType` fragments the data into smaller groups, for example :

```
| mobile | application/font-woff     | NULL             |        95511 |   164404 |    0.5809530181747403 |
| mobile | application/font-woff2    | NULL             |      1384667 |  1394934 |    0.9926397951444298 |
| mobile | image/gif                 | NULL             |     33980072 | 36297076 |    0.9361655467784787 |
| mobile | image/jpeg                | NULL             |     20143073 | 20341833 |    0.9902290024699347 |
```

Reply:
I suggested `type` rather than `mimetype`. It groups them more: https://github.com/HTTPArchive/data-pipeline/blob/08cabedf000dfea42ea8ececebe7adc491224866/modules/utils.py#L51